include/cuco/detail/extent/extent.inl

```diff
@@ -31,8 +31,7 @@ template <typename SizeType, std::size_t N>
 struct window_extent {
   using value_type = SizeType;  ///< Extent value type
 
-  __host__ __device__ constexpr value_type value() const noexcept { return N; }
-  __host__ __device__ explicit constexpr operator value_type() const noexcept { return value(); }
+  __host__ __device__ explicit constexpr operator value_type() const noexcept { return N; }
```

Comment:
I would keep the `value()` member function for the case where the `value_type` isn't known. For this case, downcasting would look like this: `static_cast<decltype(typename extent_type::value_type)>(extent)` which is a mouthful compared to `extent.value()`